components/camel-mongodb3/src/main/java/org/apache/camel/component/mongodb3/MongoDbEndpoint.java

```diff
@@ -289,7 +290,9 @@ public Exchange createMongoDbExchange(Document dbObj) {
     
     @Override
     protected void doStart() throws Exception {
-        mongoConnection = CamelContextHelper.mandatoryLookup(getCamelContext(), connectionBean, MongoClient.class);
+        if( mongoConnection == null ) {
+            mongoConnection = CamelContextHelper.mandatoryLookup(getCamelContext(), connectionBean, MongoClient.class);
+        }
         log.debug("Resolved the connection with the name {} as {}", connectionBean, mongoConnection);
```

Comment:
this is misleading as the `mongoConnection` is now unrelated from the `connectionBean` parameter

Reply:
Maybe creating a new URIParam beside mongoConnection? but in that way we may end up having 2 connection beans... I don't like that either. What if we get the `mongoConnection` as originally was and if that is null "fallback" to the one set by properties? I'd like to make backward compatible change and leave default behavior as it was before the change.

Reply:
I would say the opposite so your implementation is correct. 
My reasoning is that assuming I have a statement like:

    to("mongodb:test?mongoConnection=#myConnection")

then I do expect that the endpoint uses `myConnection` as I explicit set it. Doing the opposite is going to cause some headhake.

My comment was more about the log entry that could be wrong in case the connections comes from the `mongoConnection` parameter :-)